openfisca_france/model/prestations/aides_logement.py

```diff
@@ -1178,54 +1156,166 @@ class aides_logement_primo_accedant_k(Variable):
-        multi_n = parameters(period).prestations.al_param_accal.multiplicateur_de_n
-        R = famille('aides_logement_primo_accedant_ressources', period)
-        N = famille('aides_logement_primo_accedant_nb_part', period)
+        param = parameters(period).prestations.al_param_accal
+
+        coef_k = param.constante_du_coefficient_k
+        multi_n = param.multiplicateur_de_n
+
+        R = famille('aide_logement_base_ressources', period)
+        N = famille('aides_logement_nb_part', period)
+
+        return coef_k - (R / (multi_n * N))
+
+
+class aides_logement_foyer_k_al(Variable):
+    value_type = float
+    entity = Famille
+    label = u"Allocation logement pour les logements foyers loyer minimal"
+    reference = [u"https://www.legifrance.gouv.fr/affichCodeArticle.do?cidTexte=LEGITEXT000006074096&idArticle=LEGIARTI000006898932&dateTexte=&categorieLien=cid",
+                 u"https://www.legifrance.gouv.fr/affichCodeArticle.do?cidTexte=LEGITEXT000006073189&idArticle=LEGIARTI000006737341&dateTexte=&categorieLien=cid"]
+    definition_period = MONTH
+
+    def formula(famille, period, parameters):
+        param = parameters(period).prestations.al_param_accal
+        param2 = parameters(period).prestations.al_param
+
+        coef_k = param.constante_du_coefficient_k
+        multi_n = param2.multiplicateur_de_n_dans_la_formule_de_k
+
+        R = famille('aide_logement_base_ressources', period)
+        N = famille('aides_logement_nb_part', period)
 
         return coef_k - (R / (multi_n * N))
 
 
-class aides_logement_primo_accedant_nb_part(Variable):
+class aides_logement_foyer_k_apl(Variable):
+    value_type = float
+    entity = Famille
+    label = u"Allocation logement pour les logements foyers loyer minimal"
+    reference = [u"https://www.legifrance.gouv.fr/affichCodeArticle.do?cidTexte=LEGITEXT000006074096&idArticle=LEGIARTI000006898932&dateTexte=&categorieLien=cid",
+                 u"https://www.legifrance.gouv.fr/affichCodeArticle.do?cidTexte=LEGITEXT000006073189&idArticle=LEGIARTI000006737341&dateTexte=&categorieLien=cid"]
+    definition_period = MONTH
+
+    def formula(famille, period, parameters):
+        param = parameters(period).prestations.al_param_accal
+        param1 = parameters(period).prestations.al_param_accapl.multiplicateur_de_n
+
+        r_apl1 = param1.dans_la_formule_de_kl_numerateur
+
+        coef_k = param.constante_du_coefficient_k_apl1
+        multi_n = param1.dans_la_formule_de_kl
+
+        R = famille('aide_logement_base_ressources', period)
+        N = famille('aides_logement_nb_part', period)
+
+        return coef_k - (((R - (r_apl1 * N))) / (multi_n * N))
+
+
+class aides_logement_categorie(Variable):
+    value_type = str
+    entity = Famille
+    definition_period = MONTH
+
+    def formula(famille, period, parameters):
+        categorie_apl = famille.demandeur.menage('logement_conventionne', period)
+        return array(['al', 'apl'])[1 * categorie_apl]
```

Comment:
Still confused why we are not just using `where` here. 

`where(categorie_apl, 'apl', 'al')` gives strictly the same result, is more straightforward, consistent with the rest of the codebase, and doesn't need an unusual import.